--- ftn_stock_patched/agioncmd.ftn90	2022-08-31 17:44:20.567472848 -0300
+++ ftn_msl/agioncmd.ftn90	2020-12-08 16:02:33.892453518 -0300
@@ -268,6 +268,7 @@
 
             ! global attributes
             call nccheck ( nf90_put_att( ncid, NF90_GLOBAL, 'Conventions', 'CF-1.5') )
+            call nccheck ( nf90_put_att( ncid, NF90_GLOBAL, 'Institution', 'MetOcean Solutions Ltd') )
             call nccheck ( nf90_put_att( ncid, NF90_GLOBAL, 'History', history) )
             call nccheck ( nf90_put_att( ncid, NF90_GLOBAL, 'Directional_convention', dconv) )
 
@@ -378,7 +379,8 @@
             call nccheck ( nf90_def_dim( ncid, 'frequency', spcgrid%nfreq, spcgrid%frq_dimid ) );
             call nccheck ( nf90_def_var( ncid, 'frequency', NF90_FLOAT, spcgrid%frq_dimid, spcgrid%frq_varid ) );
             call nccheck ( nf90_put_att( ncid, spcgrid%frq_varid, 'units', 's-1') )
-            call nccheck ( nf90_put_att( ncid, spcgrid%frq_varid, 'standard_name', 'wave_frequency') )
+            call nccheck ( nf90_put_att( ncid, spcgrid%frq_varid, 'long_name', 'sea surface wave frequency') )                              !MSL (added long name)
+            call nccheck ( nf90_put_att( ncid, spcgrid%frq_varid, 'standard_name', 'sea_surface_wave_frequency') )                          !MSL (using CF standard name)
 
             call nccheck ( nf90_put_att( ncid, spcgrid%frq_varid, 'flow',  spcgrid%frequency(1) ) )
             call nccheck ( nf90_put_att( ncid, spcgrid%frq_varid, 'fhigh', spcgrid%frequency(spcgrid%nfreq) ) )
@@ -393,10 +395,11 @@
                 call nccheck ( nf90_def_dim( ncid, 'direction', spcgrid%ndir, spcgrid%dir_dimid ) )
                 call nccheck ( nf90_def_var( ncid, 'direction', NF90_FLOAT, spcgrid%dir_dimid, spcgrid%dir_varid ) );
                 call nccheck ( nf90_put_att( ncid, spcgrid%dir_varid, 'units', 'radians') )
-                call nccheck ( nf90_put_att( ncid, spcgrid%dir_varid, 'long_name', 'direction') )
                 if ( nautical ) then
+                    call nccheck ( nf90_put_att( ncid, spcgrid%dir_varid, 'long_name', 'sea surface wave from direction') )                 !MSL (add convention to long name)
                     call nccheck ( nf90_put_att( ncid, spcgrid%dir_varid, 'standard_name', 'sea_surface_wave_from_direction') )
                 else
+                    call nccheck ( nf90_put_att( ncid, spcgrid%dir_varid, 'long_name', 'sea surface wave from direction') )                 !MSL (add convention to long name)
                     call nccheck ( nf90_put_att( ncid, spcgrid%dir_varid, 'standard_name', 'sea_surface_wave_to_direction') )
                 end if
                 call nccheck ( nf90_put_att( ncid, spcgrid%dir_varid, 'mdc',   spcgrid%ndir ) )
@@ -417,7 +420,7 @@
                 else
                     call nccheck ( nf90_put_att( ncid, evarid, '_FillValue',NF90_FILL_FLOAT) )
                 end if
-                call nccheck ( nf90_put_att( ncid, evarid, 'long_name', 'density') )
+                call nccheck ( nf90_put_att( ncid, evarid, 'long_name', 'sea surface wave directional variance spectral density') )         !MSL (updated long name)
                 call nccheck ( nf90_put_att( ncid, evarid, 'units', 'm2 s rad-1') )
                 call nccheck ( nf90_put_att( ncid, evarid, 'standard_name', &
                             'sea_surface_wave_directional_variance_spectral_density') )
